pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -174,6 +184,16 @@ func LocationPrioritizedComparison(currentLeaderRevision string, l *LeaderElecti
 	return l.revision == currentLeaderRevision && !l.remote && currentLeaderRemote
 }
 
+// Enabled sets whether leader election is enabled. Calling Enabled(false)
+// bypasses leader election and assumes that we are always leader, avoiding
+// unnecessary lease updates on single-node clusters.
+//
+// It is not thread-safe and must be called before Run().
+func (l *LeaderElection) Enabled(enabled bool) *LeaderElection {
+	l.enabled = enabled
```

Comment:
I would prefer passing as a parama, then we will not need to init watcher.

Reply:
Done.